indigo_app/static/javascript/indigo/views/document_editor.js

```diff
@@ -672,37 +671,28 @@
       this.textEditor.focus();
     },
 
-    getAnnotationAmendingWork: function(document) {
+    getAmendingWork: function(document) {
       var date = document.get('expression_date'),
-          documentAmendments = document.get('amendments'),
-          amendments = [];
+          documentAmendments = Indigo.Preloads.amendments,
+          amendment = _.findWhere(documentAmendments, {date: date});
 
-      documentAmendments.forEach(function(amendment) {
-        if (amendment.date === date) {
-          amendments.push(amendment);
-        }
-      });
-
-      if (amendments) {
-        return amendments[0];
+      if (amendment) {
+        return amendment.amending_work;
       }
 
     },
 
-    insertAnnotation: function(e) {
+    insertRemark: function(e) {
       e.preventDefault();
-      var element = 'section/subsection/paragraph/subparagraph/<other> ()',
-          verb = 'substituted/amended/inserted/deleted/repealed/<other>',
-          amendingSection = 'section <insert>',
-          annotation;
+      var verb = e.currentTarget.className.split(" ").pop(),
```

Comment:
This is very fragile. Much better to put the data into a data attribute (eg. `<span ... data-verb="foo">`) and use `e.currentTarget.getAttribute('data-verb')`.